core/domain/prod_validation_jobs_one_off.py

```diff
@@ -0,0 +1,141 @@
+# coding: utf-8
+#
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for validating prod models."""
+from core import jobs
+from core.platform import models
+
+(
+    user_models, exp_models, collection_models,
+    feedback_models) = models.Registry.import_models([
+        models.NAMES.user, models.NAMES.exploration,
+        models.NAMES.collection, models.NAMES.feedback])
+datastore_services = models.Registry.import_datastore_services()
+
+
+def has_model_for_id(model_class, model_id):
+    """Checks there's an existing nondeleted model for given model_id.
+
+    Args:
+        model_class: BaseModel. Class of the model.
+        model_id: str. Key of the model.
+
+    Returns:
+        bool. True if there exists a model with given model_id.
+        """
+    model = model_class.get_by_id(model_id)
+    return model is not None and model.deleted is False
+
+
+class UserSubscriptionsModelValidator(object):
+    """Class for validating UserSubscriptionModels."""
+
+    errors = []
+    checks_passed = []
+
+    @classmethod
+    def validate(cls, item):
+        """Validates a UserSubscriptionsModel entity.
+
+        Args:
+            item: UserSubscriptionsModel. Entity to validate.
+
+        Returns:
+            string. Validation status.
+        """
+        # For each key, checks the entries in that field of
+        # UserSubscriptionsModel should be valid non-deleted instances of the
+        # given model classes.
+        external_ids_checks = {
+            'activity_ids': exp_models.ExplorationModel,
+            'collection_ids': collection_models.CollectionModel,
+            'general_feedback_thread_ids': (
+                feedback_models.GeneralFeedbackThreadModel),
+            'creator_ids': user_models.UserSettingsModel,
+        }
+        if not has_model_for_id(user_models.UserSettingsModel, item.id):
+            cls.errors.append(
+                'id %s is not valid %s' % (
+                    item.id,
+                    user_models.UserSettingsModel.__name__))
+        else:
+            cls.checks_passed.append(['id check passed'])
+        cls._check_external_ids(item, external_ids_checks=external_ids_checks)
+
+    @classmethod
+    def _check_external_ids(cls, item, **kwargs):
```

Comment:
Try to avoid using **kwargs; that's a bit of an antipattern. Specify things explicitly. (But see above note on rearchitecture of the "external IDs" constant.)